src/app/rpmostree-dbus-helpers.cxx

```diff
@@ -1061,12 +1062,13 @@ rpmostree_sort_pkgs_strv (const char *const* pkgs,
           g_str_has_prefix (*pkg, "https://"))
         {
           g_print ("Downloading '%s'... ", *pkg);
-          glnx_autofd int fd = ror_download_to_fd (*pkg, error);
-          if (fd < 0)
-            {
-              g_print ("failed!\n");
-              return FALSE;
-            }
+          glnx_autofd int fd = -1;
+          try {
+            fd = rpmostreecxx::download_to_fd(*pkg);
+          } catch (std::exception& e) {
```

Comment:
This one is a really interesting topic.  `GError` closer to rust's explicit `Result` than C++ exceptions are.  However...for *most* of the usage of `GError` in rpm-ostree we are effectively just using it to unwind up the stack - i.e. we don't explicitly catch errors.

The only exceptions I can think of are:

 - `main()`  (obviously)
 - GDBus method handlers

And so what I'd probably argue for is that rather than doing try/catch-to-gerror at places like this we just go with the C++ exception flow, and install toplevel try/catch at those places that need it.

(Also in rpm-ostree the usage of `g_error_matches()` is something like 90% for ENOENT...which totally validates https://github.com/cgwalters/openat-ext/blob/26324991dd1648bab64ce1f2e6471186e6ed21f9/src/lib.rs#L36 )

Reply:
Now that `main()` has a `try/catch`, I think can drop this one, right?